vendor/golang.org/x/sys/unix/zptrace_mipsnn_linux.go

```diff
@@ -42,10 +42,10 @@ type PtraceRegsMips64 struct {
 
 // PtraceGetRegsMips64 fetches the registers used by mips64 binaries.
 func PtraceGetRegsMips64(pid int, regsout *PtraceRegsMips64) error {
-	return ptrace(PTRACE_GETREGS, pid, 0, uintptr(unsafe.Pointer(regsout)))
+	return ptracePtr(PTRACE_GETREGS, pid, 0, unsafe.Pointer(regsout))
 }
 
 // PtraceSetRegsMips64 sets the registers used by mips64 binaries.
 func PtraceSetRegsMips64(pid int, regs *PtraceRegsMips64) error {
-	return ptrace(PTRACE_SETREGS, pid, 0, uintptr(unsafe.Pointer(regs)))
+	return ptracePtr(PTRACE_SETREGS, pid, 0, unsafe.Pointer(regs))
 }
```

Comment:
The code patch updates the use of `ptrace` function in two functions `PtraceGetRegsMips` and `PtraceSetRegsMips` for mips binaries and also for two corresponding functions `PtraceGetRegsMips64` and `PtraceSetRegsMips64` for mips64 binaries, to use a new function `ptracePtr` instead. The new function seems to have changed the fourth argument from `uintptr(unsafe.Pointer(regs))` to `unsafe.Pointer(regs)`. This change should not introduce any bug risk as it's just a different way of passing a pointer to C API. It could potentially improve code readability by avoiding the non-standard type conversion.